src/storage/src/flush.rs

```diff
@@ -160,6 +164,8 @@ pub struct FlushJob<S: LogStore> {
     pub wal: Wal<S>,
     /// Region manifest service, used to persist metadata.
     pub manifest: RegionManifest,
+    /// Callbacks that get invoked on flush success.
+    pub on_success: tokio::sync::Mutex<Option<FlushCallback>>,
```

Comment:
Seems we don't need the mutex.

Reply:
Without the mutex the FlushJob will not be `Sync` so referencing `self` across await point won't be possible when calling `run`:
https://github.com/GreptimeTeam/greptimedb/blob/b4e7dcb74ee30147385c6ae447e82e401616395d/src/storage/src/flush.rs#L247-L255